src/foam/java/refinements.js

```diff
@@ -156,14 +165,13 @@ foam.CLASS({
     function generateSetter_() {
       return this.javaSetter ? this.javaSetter : `
         if ( this.__frozen__ ) throw new UnsupportedOperationException("Object is frozen.");
-        assert${foam.String.capitalize(this.name)}(val);
-        ${this.name}_ = val;
-        ${this.name}IsSet_ = true;
+        setProperty(\"${this.name}\", val);
       `;
     },
 
     function buildJavaClass(cls) {
       if ( ! this.generateJava ) return;
+      if ( foam.java.Enum.isInstance(cls) ) return;
```

Comment:
You have just removed setters for Enums in Java, you've completely disabled properties for enums in java.  This is now inconsistent with JS where we allow properties because often an Enum contains data you want to read.

I think it's better to continue to allow properties for Enums and just make them final by default, either by adding final support as necessary or initializing enums as frozen.